packages/page-staking/src/Performance/Performance.tsx

```diff
@@ -0,0 +1,242 @@
+  const [isPalletElectionsSupported, setIsPalletElectionsSupported] = useState<boolean | undefined>(undefined);
+
+  const [sessionValidatorBlockCountLookup, setSessionValidatorBlockCountLookup] = useState<[string, number][]>([]);
+
+  const MINIMUM_SUPPORTED_ELECTIONS_PALLET_VERSION = 3;
+
+  const eraExposure = useCall<DeriveEraExposure>(api.derive.staking.eraExposure, [sessionEra.era]);
+
+  const eraValidators = useMemo(() => {
+    if (eraExposure?.validators) {
+      return Object.keys(eraExposure?.validators);
+    }
+
+    return [];
+  }, [eraExposure]
+  );
+
+  useEffect(() => {
+    const sessionPeriod = Number(api.consts.elections.sessionPeriod.toString());
+    const firstBlockInSession = sessionEra.session * sessionPeriod;
+
+    api.rpc.chain
+      .getBlockHash(firstBlockInSession)
+      .then((result): void => {
+        setFirstBlockInSessionHash(result);
+      })
+      .catch(console.error);
+  },
+  [api, sessionEra]
+  );
+
+  useEffect(() => {
+    if (firstBlockInSessionHash !== undefined) {
+      api.at(firstBlockInSessionHash.toString()).then((result) => {
+        result.query.elections.palletVersion().then((version) => {
+          setIsPalletElectionsSupported(Number(version.toString()) >= MINIMUM_SUPPORTED_ELECTIONS_PALLET_VERSION);
+        }).catch(console.error);
+      }).catch(console.error);
+    }
+  }, [api, firstBlockInSessionHash]);
+
+  useEffect(() => {
+    if (!sessionEra.currentSessionMode) {
+      const sessionPeriod = Number(api.consts.elections.sessionPeriod.toString());
+      const lastBlockInSession = (sessionEra.session + 1) * sessionPeriod - 1;
+
+      api.rpc.chain
+        .getBlockHash(lastBlockInSession)
+        .then((result): void => {
+          setLastBlockInSessionHash(result);
+        })
+        .catch(console.error);
+    }
+  },
+  [api, sessionEra]
+  );
+
+  useEffect(() => {
+    if (firstBlockInSessionHash !== undefined) {
+      api.derive.chain.getHeader(firstBlockInSessionHash).then((header) => {
+        if (header && !header.isEmpty && header.author) {
+          setFirstSessionBlockAuthor(header.author.toString());
+        }
+      }).catch(console.error);
+    }
+  },
+  [api, firstBlockInSessionHash]
+  );
+
+  function parseSessionBlockCount (sessionValidatorBlockCountValue: [StorageKey<AnyTuple>, Codec][], firstSessionBlockAuthor: string): [string, number][] {
+    return sessionValidatorBlockCountValue.map(([key, values]) => {
+      const account = key.args[0].toString();
+      let count = Number(values.toString());
+
+      if (account === firstSessionBlockAuthor) {
+        // a workaround for the fact that the first session block author is not reflected in that block
+        // elections.sessionValidatorBlockCount state
+        count += 1;
+      }
+
+      return [account, count];
+    });
+  }
+
+  useEffect(() => {
+    if (lastBlockInSessionHash !== undefined && firstSessionBlockAuthor !== undefined) {
```

Comment:
if you only want to check if these variables have non-empty value, consider `if (lastBlockInSessionHash && firstSessionBlockAuthor) {...}` (or sometimes it's safer to check it like this: `if (!!lastBlockInSessionHash && !!firstSessionBlockAuthor) {...}`)

Reply:
Of course if you want to make this condition true if any of these two have the value of 0, this can be obtained the way it is currently done.

Reply:
Good point, done